Fix an issue with cwiseMax() and eigen3 < 3.2.4
Display how bad the covariance matrix was when it becomes not positive definite.

diff --git codels/ukf_filter.cc codels/ukf_filter.cc
index 9b0747c..3bde6c2 100644
--- codels/ukf_filter.cc
+++ codels/ukf_filter.cc
@@ -16,6 +16,9 @@
  */
 #include <err.h>
 
+#include <sstream>
+#include <string>
+
 #include "ukf.h"
 
 #include <Eigen/Geometry>
@@ -83,9 +86,16 @@ class filter_s::impl_s
       warnx("state covariance is not positive definite ... fixing");
 
       Eigen::EigenSolver<state_s::dcov_s> e(cov);
-      state_s::dcov_s d = e.eigenvalues().real().cwiseMax(1e-10).asDiagonal();
+      state_s::dcov_s d = e.eigenvalues().real().asDiagonal();
       state_s::dcov_s v = e.eigenvectors().real();
 
+      {
+        std::ostringstream msg;
+        msg << "diagonal was [" << d.diagonal().transpose() << "]'";
+        warnx("%s", msg.str().c_str());
+      }
+      d.diagonal() =
+        d.diagonal().cwiseMax(state_s::delta_s::Constant(1e-10));
       cov = v * d * v.inverse();
 
       llt.compute((state_s::dof + lambda()) * cov);
